### PR TITLE
OCGV - Fixes OutputMode Single

### LINKED ISSUE #23: -OutPutMode not honored
Name             : ConsoleHost
Version          : 6.2.1

ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Script     0.1.1      Microsoft.PowerShell.GraphicalTools Core      {Out-GridView, ogv}

Test was trying to select a single row for further processing.

### PR DESCRIPTION
**Pull request by tig**
This PR fixes #23 and #79 for `ocgv`. It is dependent on PR #106.

Previously `-OutputMode Single` worked exactly like `-OutputMode Multiple` which was a bug. With this PR all three modes (`None`, `Single`, and `Multiple`) work correctly:

* *None* - No marking of items is possible. ENTER does nothing. ESC cancels.
* *Single* - No marking of multiple items is possible. Single selection 'mark UI' is shown. SPACE selects a single item. NTER selects the selected item and it is returned. ESC cancels.
* *Multiple* - Marking of multiple items is possible with Multiple selection 'mark UI' using SPACE. ENTER returns selected items. ESC cancels.

* Closes: #79

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
I see a couple changes from the other PR. Can you rebase?

File: src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public HashSet<int> Start(ApplicationData applicationData)
             _applicationData = applicationData;
             _gridViewDetails = new GridViewDetails
             {
-                // If we have an OutputMode, then we want to make them selectable. If we make them selectable,
+                // If OutputMode is Multiple, then we make items selectable. If we make them selectable,
                 // they have a 8 character addition of a checkbox ("     [ ]") that we have to factor in.
                 ListViewOffset = _applicationData.OutputMode != OutputModeOption.None ? 8 : 4
```
Comment (TylerLeonhardt) [inline src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs:32]: don't you need to change this?

Comment (tig) [reply src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs:32]: The spacing is correct for both `Single` and `Multiple` modes now (we are now using `Terminal.Gui`'s `AllowsMultipleSelection` feature.

the comment is not quite right. I'll fix it.

**Comment by TylerLeonhardt** [comment]
I don't know why anyone would use `None` at this point... but I'm fine keeping it in for consistancy

**Review by TylerLeonhardt — APPROVED** [review]
LGTM

**Pull request merged**